clang/lib/CodeGen/BackendUtil.cpp

```diff
@@ -995,6 +996,13 @@ void EmitAssemblyHelper::RunOptimizationPipeline(
             FPM.addPass(BoundsCheckingPass());
           });
 
+    if (LangOpts.Sanitize.has(SanitizerKind::Realtime)) {
```

Comment:
Now that we have the llvm work done (in the earlier commit), we now check if the sanitizer is enabled, and insert this "Pass" that we defined in LLVM.

As for where it should live `registerScalarOptimizerLateEPCallback` was my guess, as it says it is "very late in the pipeline" but we likely have to ask reviewers what is appropriate